samples/matter/weather_station/CMakeLists.txt

```diff
@@ -6,6 +6,12 @@
 
 cmake_minimum_required(VERSION 3.13.1)
 
+if (NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE release)
```

Comment:
I think that we are already not consistent, because other samples use overlays instead of two types of prj.confs, so generally I can do that, but I don't think that it will change that much.

Reply:
Perhaps, they use ZDebug or ZRelease as Debug and Release are known CMake values (https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html) and some build scripts may set different options for them. Not sure if "debug" and "release" make a difference. :)
Btw, using overlays vs prj_xxx.conf is hidden to a normal user so I wouldn't worry about that inconsistency.
That being said, I don't have a strong opinion on the choice, just throwing some arguments :).

Reply:
As discussed F2F we decided to consider changing naming by the way of moving sample to the applications.